managers/neo4j.go

```diff
@@ -27,28 +29,42 @@ func (m *Neo4jManager) Close() {
 	m.driver.Close()
 }
 
-func (m *Neo4jManager) CreateOrUpdateNode(userSummary *protos.UserSummary) error {
-	// TODO: this does not update right now.
-	// Must figure out how to update the data if userid already exists.
-	_, err := m.executeQuery(fmt.Sprintf(`
-			MERGE ({
-				userid: "%v",
-				deviceid: "%v",
-				name: "%v",
-				birthdate: %v,
-				thumbnail: "%v"
-			})
-		`, userSummary.UserId, userSummary.DeviceId, userSummary.Name, userSummary.Birthdate, userSummary.ThumbnailUri))
+func (m *Neo4jManager) CreateOrUpdateUserNode(user *protos.User) error {
+	userModel := usermodel.UserModel{User: user}
+	q := fmt.Sprintf(`
+		MERGE (user { userId : "%[1]v" })
+		ON CREATE SET user = %[2]v
+		ON MATCH SET user +=  %[2]v
+		`, userModel.UserId, userModel.String())
+	_, err := m.executeQuery(q)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
+func (m *Neo4jManager) FetchUserNode(userID string) (*protos.User, error) {
+	records, err := m.executeQuery(fmt.Sprintf(`
+			MATCH p = ({userId: "%v"})
+			UNWIND NODES(p) as userNode
+			RETURN userNode{.*}
+		`, userID))
+	if err != nil {
+		return nil, err
+	}
+	if len(records) > 1 {
+		return nil, errors.New("multiple user nodes found")
+	}
+	if len(records) == 0 {
+		return &protos.User{}, nil
```

Comment:
```suggestion
		return nil, nil
```
Think that would be easier for us to check on FE that the account doesn't exist yet.